controller-server/src/main/java/com/yahoo/vespa/hosted/controller/athenz/filter/AthenzPrincipalFilter.java

```diff
@@ -81,8 +82,9 @@ public void filter(DiscFilterRequest request, ResponseHandler responseHandler) {
     }
 
     private static Optional<X509Certificate> getClientCertificate(DiscFilterRequest request) {
-        return Optional.ofNullable((X509Certificate[]) request.getAttribute("jdisc.request.X509Certificate"))
-                .map(chain -> chain[0]);
+        List<X509Certificate> chain = request.getClientCertificateChain();
```

Comment:
Could be simplified to `request.getClientCertificateChain().stream().findFirst()`.

Reply:
Thanks for the tip. Completely forgot that one :)